src/gluonts/model/naive_2/_predictor.py

```diff
@@ -0,0 +1,153 @@
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.evaluation import get_seasonality
+import statsmodels.api as sm
+
+
+def seasonality_test(past_ts_data: np.array, season_length: int):
+    """
+    Test the time-series for seasonal patterns by performing a 90% auto-correlation test:
+
+    As described here: https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+    Code based on: https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R
+    """
+    critical_z_score = 1.645  # corresponds to 90% confidence interval
+    if len(past_ts_data) < 3 * season_length:
+        is_seasonal = False
+    else:
+        # calculate auto-correlation for lags up to season_length
+        auto_correlations = sm.tsa.stattools.acf(
+            past_ts_data, fft=False, nlags=season_length
+        )
+        auto_correlations[1:] = 2 * auto_correlations[1:] ** 2
+        limit = (
+            critical_z_score
+            / np.sqrt(len(past_ts_data))
+            * np.sqrt(np.cumsum(auto_correlations))
+        )
+        is_seasonal = (
+            abs(auto_correlations[season_length]) > limit[season_length]
+        )
+
+    return is_seasonal
+
+
+def naive_02(past_ts_data: np.array, forecast_length: int, freq: str):
+    """
+    Make seasonality adjusted time series prediction.
+
+    As described here: https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+    Code based on: https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R
+    """
+    season_length = get_seasonality(freq)
+    has_seasonality = False
+
+    if season_length > 1:
+        has_seasonality = seasonality_test(past_ts_data, season_length)
+
+    # it has seasonality, then calculate the multiplicative seasonal component
+    if has_seasonality:
+        # TODO: think about maybe only using past_ts_data[- max(5*season_length, 2*forecast_length):] for speedup
+        seasonal_decomposition = sm.tsa.seasonal_decompose(
+            x=past_ts_data, period=season_length, model="multiplicative"
+        ).seasonal
+        seasonality_normed_context = past_ts_data / seasonal_decomposition
+
+        last_period = seasonal_decomposition[-season_length:]
+        num_required_periods = (forecast_length // len(last_period)) + 1
+
+        multiplicative_seasonal_component = np.tile(
+            last_period, num_required_periods
+        )[:forecast_length]
+    else:
+        seasonality_normed_context = past_ts_data
+        multiplicative_seasonal_component = np.ones(
+            forecast_length
+        )  # i.e. no seasonality component
+
+    # calculate naive 1 forecast: (last value forecast_length times)
+    naive_1_forecast = (
+        np.ones(forecast_length) * seasonality_normed_context[-1]
+    )
+
+    forecast = np.mean(naive_1_forecast) * multiplicative_seasonal_component
+
+    return forecast
```

Comment:
Do you think you could maybe take a look at this code and give feedback whether its true to your definition of the Naive 2 model? @Mcompetitions

Reply:
Ok, so we get the exact same results, so I think we're good.